tls/s2n_handshake_io.c

```diff
@@ -567,6 +582,47 @@ static int s2n_handshake_handle_sslv2(struct s2n_connection *conn)
     return 0;
 }
 
+/* Sometimes the handshake state machine can be blocked on application data. 
+ * For example, if the s2n server is using a remote upstream peer for session cache lookup,
+ * the handshake state machine will be blocked until there is data returned from the 
+ * session cache server. In such a case, there is no data in the underlying io, 
+ * so we can just call the corresponding handler to process the application data.
+ * Note that we need a return value 1 to indicate that app data is not yet avaiable 
+ * and the state machine is still blocked; a return value 0 to indicate the app data is 
+ * successfully processed and state machine advanced; -1 to indicate an error and 
```

Comment:
Do we need a named constants for -1, 0, 1 to improve readability?

Reply:
agree we should use named constants, but since there are many places using -1 and 0 already. If we use named constants, we'd better change it everywhere. So it makes more sense to make this change in a separate PR with a dedicated purpose of refactoring this.